pbs/usersync.go

```diff
@@ -71,20 +73,27 @@ type UserSyncDeps struct {
 }
 
 // ParsePBSCookieFromRequest parses the UserSyncMap from an HTTP Request.
-func ParsePBSCookieFromRequest(r *http.Request) *PBSCookie {
-	cookie, err := r.Cookie(COOKIE_NAME)
-	if err != nil {
+func ParsePBSCookieFromRequest(r *http.Request, configuredOptoutCookie *config.Cookie) *PBSCookie {
```

Comment:
In `pbs_light.go`, you've got:

> userSyncCookie := pbs.ParsePBSCookieFromRequest(r, c.OptOutCookie)
> userSyncCookie.SetCookieOnResponse(w, hostCookieSettings.Domain)

I thought that block was odd, at first... but given this code block, it does make sense. It keeps the PBS cookie data synchronized with the Host cookie.

On the other hand... it's not very efficient to set the cookie on the response if the PBS cookie's values are already in sync. So... I've got two ideas which would improve this. Not sure which one you find cleaner:

1. Return `(*PBSCookie, bool)` here, where the `bool` tells "is this data in sync with the PBS cookie?"
2. Add a `func (cookie *PBSCookie) hasChanged() bool` function, and make all the mutators set it to `true` 

I'm partial to (2)... but both seem fine to me.